tests/test_sft_trainer.py

```diff
@@ -487,6 +496,58 @@ def test_train_dense_with_peft_config(self):
             elif "base_layer" not in n:  # We expect the peft parameters to be different (except for the base layer)
                 self.assertFalse(torch.allclose(param, new_param), f"Parameter {n} has not changed")
 
+    @parameterized.expand(
```

Comment:
when running the CI without optional dependencies (thus without peft), this will fail because it tries to use something which is not imported (`PromptEncoderConfig`, ...)

Instead, I recommend something like:

```python
@parameterized.expand([("prompt_encoder",), ...])
@require_peft
def test_train_with_peft_config_prompt_tuning(self, peft_case):
    if peft_case == "prompt_encoder":
        peft_config = PromptEncoderConfig(task_type=TaskType.CAUSAL_LM, num_virtual_tokens=4)
        gradient_checkpointing = True
    elif:
        ...